nixos/modules/virtualisation/vmware-host.nix

```diff
@@ -0,0 +1,191 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.virtualisation.vmware.host;
+  wrapperDir = "/run/vmware/bin"; # Perfectly fits as /usr/local/bin
+  parentWrapperDir = dirOf wrapperDir;
+  vmwareWrappers = # Needed as hardcoded paths workaround
+  let mkVmwareSymlink =
+  program:
+  ''
+    ln -s "${config.security.wrapperDir}/${program}" $wrapperDir/${program}
+  '';
+  in
+  [
+    (mkVmwareSymlink "pkexec")
+    (mkVmwareSymlink "mount")
+    (mkVmwareSymlink "umount")
+    (mkVmwareSymlink "vmware-vmx")
+    (mkVmwareSymlink "vmware-gksu")
+  ];
+
+in
+{
+  options = {
+    virtualisation.vmware.host = {
+      enable = mkEnableOption "Enable VMWare host virtualisation";
+      package = mkOption {
+        type = types.package;
+        default = pkgs.vmware-workstation;
+        defaultText = literalExpression "pkgs.vmware-workstation";
+        description = ''VMWare host virtualisation package to use
+        <note><para>
+        <literal>vmware-vmx</literal> will likely cause CPU hogging kcompactd0 process to appear because of Transparent Hugepages
+        Use <literal>boot.kernelParams = [ "transparent_hugepage=never" ];</literal> to disable them
+        If that didn't work disable <literal>TRANSPARENT_HUGEPAGE</literal> and <literal>COMPACTION</literal> config option when compiling kernel
+        </para></note>
+        '';
+      };
+      extraPackages = mkOption {
+        type = with types; listOf package;
+        default = with pkgs; [];
+        defaultText = literalExpression ''
+          with pkgs; [ ]
+        '';
+        description = ''
+          Extra packages to be used with VMWare host.
+          For example <literal>ntfs3g</literal> can be added for mounting VDisks with NTFS
+        '';
+      };
+      extraConfig = mkOption {
+        type = types.lines;
+        default = "";
+        description = ''
+          Add extra config to /etc/vmware/config
+          For example to allow unsupported device's OpenGL and Vulkan acceleration for guest vGPU
+          <literal>
+          mks.gl.allowUnsupportedDrivers = "TRUE"
+          mks.vk.allowUnsupportedDevices = "TRUE"
+          </literal>
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    boot.extraModulePackages = [ config.boot.kernelPackages.vmware ];
+    boot.extraModprobeConfig = "alias char-major-10-229 fuse";
+    boot.kernelModules = [ "vmw_pvscsi" "vmw_vmci" "vmmon" "vmnet" "fuse" ];
+
+    environment.systemPackages = [ cfg.package ] ++ cfg.extraPackages;
+    services.printing.drivers = [ cfg.package ];
+
+    environment.etc."vmware/config".text = ''
+    ${builtins.readFile "${cfg.package}/etc/vmware/config"}
+    ${cfg.extraConfig}
+    '';
+
+    environment.etc."vmware/bootstrap".source = "${cfg.package}/etc/vmware/bootstrap";
+    environment.etc."vmware/icu".source = "${cfg.package}/etc/vmware/icu";
+    environment.etc."vmware-installer".source = "${cfg.package}/etc/vmware-installer";
```

Comment:
I tried to run vmware, but I get the following issue:

$ vmware
/nix/store/cck6dbcyxbx2drgvc1a2x5d7hq5g1bbj-vmware-workstation-16.2.3/bin/vmware-modconfig: line 17: /nix/store/cck6dbcyxbx2drgvc1a2x5d7hq5g1bbj-vmware-workstation-16.2.3/etc/vmware-installer/bootstrap: No such file or directory

The correct path would be:

$ ls /nix/store/cck6dbcyxbx2drgvc1a2x5d7hq5g1bbj-vmware-workstation-16.2.3/etc/vmware/
bootstrap  config     icu/

Not sure if this is something that needs to be fixed through these lines

Reply:
Looking at code this occurs if vmmon, vmnet kernel modules are not found, try using module's enable option and restart to load them. 

`vmware-modconfig` is broken binary for non-FHS, yet it's avoidable so maybe i should replace it with dummy file that prompts user to restart?

Reply:
vmware-modconfig now only displays dialog to restart, it's originally used to display kernel modules compile dialog but it's useless for NixOS anyway